templates/edit.html

```diff
@@ -6,9 +6,26 @@
   
     $('#remove-karaoke').click(function(e) {
         var new_name = $('#new_file_name').val();
-        new_name = new_name.replace(/\([Kk][Aa][Rr][Aa][Oo][Kk][Ee].*\)/g, "");
-        new_name = new_name.replace(/[Kk][Aa][Rr][Aa][Oo][Kk][Ee]/g, "");
+        new_name = new_name.replace(/\([Hh][Dd][ ].*\)/g, " ");
+        new_name = new_name.replace(/\s\s+/g, "");
+        new_name = new_name.replace(/\([Kk][Aa][Rr][Aa][Oo][Kk][Ee].*\)/g, " ");
+        new_name = new_name.replace(/\[[Kk][Aa][Rr][Aa][Oo][Kk][Ee].*\]/g, " ");
+        new_name = new_name.replace(/[Kk][Aa][Rr][Aa][Oo][Kk][Ee][  ][Vv][Ee][Rr][Ss][Ii][Oo][Nn]/g, " ");
+        new_name = new_name.replace(/[Kk][Aa][Rr][Aa][Oo][Kk][Ee]/g, " ");
+        new_name = new_name.replace(/\s\s+/g, " ");
+        new_name = new_name.replace("- -", "-");
+        new_name = new_name.replace("--", "-");
+        new_name = new_name.replace("as popularized by", "");
         new_name = new_name.trim();
+        
+        
+        
+        //title case
```

Comment:
hmm, I feel like if we're just going to titlecase the end might as well toLowerCase() before all the replace calls, to simplify them (wouldn't need to do this [Kk][Aa][Rr][Aa][Oo][Kk][Ee] business and could do simply: "karaoke" to be more readable.

Reply:
I agree. Unfortunately, I did commit to my own preferences.